src/internal/MonoColor.h

```diff
@@ -0,0 +1,209 @@
+    // Construct a MonoColor using R, G, B, W values (0-255)
+    // ------------------------------------------------------------------------
+    MonoColor(uint8_t r, uint8_t g, uint8_t b, uint8_t w = 0);
+
+    // ------------------------------------------------------------------------
+    // Construct a MonoColor using a single brightness value (0-255)
+    // This works well for creating gray tone colors
+    // (0) = black, (255) = white, (128) = gray
+    // ------------------------------------------------------------------------
+    MonoColor(uint8_t brightness);
+
+    // ------------------------------------------------------------------------
+    // Construct a MonoColor using RgbColor
+    // ------------------------------------------------------------------------
+    MonoColor(const RgbColor& color);
+
+    // ------------------------------------------------------------------------
+    // Construct a MonoColor using HtmlColor
+    // ------------------------------------------------------------------------
+    MonoColor(const HtmlColor& color);
+
+    // ------------------------------------------------------------------------
+    // Construct a MonoColor using HslColor
+    // ------------------------------------------------------------------------
+    MonoColor(const HslColor& color);
+
+    // ------------------------------------------------------------------------
+    // Construct a MonoColor using HsbColor
+    // ------------------------------------------------------------------------
+    MonoColor(const HsbColor& color);
+
+    // ------------------------------------------------------------------------
+    // Construct a MonoColor that will have its values set in latter operations
+    // CAUTION:  The R,G,B, W members are not initialized and may not be consistent
+    // ------------------------------------------------------------------------
+    MonoColor()
+    {
+    };
+
+    // ------------------------------------------------------------------------
+    // Comparison operators
+    // ------------------------------------------------------------------------
+    bool operator==(const MonoColor& other) const
+    {
+        return (W == other.W);
+    };
+
+    bool operator!=(const MonoColor& other) const
+    {
+        return !(*this == other);
+    };
+
+    // ------------------------------------------------------------------------
+    // Returns if the color is grey, all values are equal other than white
+    // ------------------------------------------------------------------------
+    bool IsMonotone() const
+    {
+        return true;
+    };
+
+    // ------------------------------------------------------------------------
+    // Returns if the color components are all zero, the white component maybe 
+    // anything
+    // ------------------------------------------------------------------------
+    bool IsColorLess() const
+    {
+        return true;
```

Comment:
IsColorless should be removed, it is types that have separate color and white channels.